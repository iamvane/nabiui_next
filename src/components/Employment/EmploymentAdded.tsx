import * as React from 'react';

import {
  Icon,
  IconButton,
  Grid,
  Typography
} from '@material-ui/core';

import { EmploymentType }  from 'components/Employment/model';
import { EmploymentAddedComponent } from 'components/Employment/constants';

interface Props extends EmploymentType {
  deleteEmployment: (employmentId: any) => void;
  editEmployment: (employmentId: any) => void;
  gridWidth?: boolean | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | undefined;
  notEditable?: boolean;
}

/**
 * EmploymentAdded
 */
const EmploymentAdded: React.StatelessComponent<Props> = props => {
  const fromMonth = props.fromMonth.charAt(0).toUpperCase() + props.fromMonth.slice(1);
  const toYear = props.toYear ? props.toYear : '';
  const toMonth = props.toMonth ? props.toMonth : '';
  const toMonthCapital = toMonth.charAt(0).toUpperCase() + toMonth.slice(1);

  return (
    <Grid
      item={true}
      md={props.gridWidth && props.gridWidth}
      xs={12}
      className={`${props.notEditable ? '' : 'nabi-editable-item'} nabi-margin-top-small`}
    >
      <div className="nabi-action-buttons">
        <IconButton
          color="primary"
          className="nabi-float-right nabi-margin-left-xsmall"
          aria-label="Delete"
          onClick={() => props.deleteEmployment(props.id)}
        >
          <Icon>delete</Icon>
        </IconButton>

        <IconButton
          color="primary"
          className="nabi-float-right"
          aria-label="Edit"
          onClick={() => props.editEmployment(props.id)}
        >
          <Icon>edit</Icon>
        </IconButton>
      </div>
      <Typography className="nabi-margin-top-small nabi-text-uppercase" variant="body2">
        <mark>{props.jobTitle}</mark>
      </Typography>
      <Typography>
        <mark>{props.employer}</mark>
      </Typography>
      {props.stillWorkHere || (!props.toYear && !toMonthCapital) ? (
        <Typography>
          <mark>
            {EmploymentAddedComponent.timelinePresent.replace(
              EmploymentAddedComponent.fromMonthPlaceholder,
              fromMonth
            ).replace(
              EmploymentAddedComponent.fromYearPlaceholder,
              props.fromYear,
            )}
          </mark>
        </Typography>
      ) : (
        <Typography>
          <mark>
            {EmploymentAddedComponent.timeline.replace(
              EmploymentAddedComponent.fromMonthPlaceholder,
              fromMonth
            ).replace(
              EmploymentAddedComponent.fromYearPlaceholder,
              props.fromYear,
            ).replace(
              EmploymentAddedComponent.toMonthPlaceholder,
              toMonthCapital,
            ).replace(
              EmploymentAddedComponent.toYearPlaceholder,
              toYear,
            )}
          </mark>
        </Typography>
      )}
      <Typography>
        <mark>{props.jobLocation}</mark>
      </Typography>
    </Grid>
  );
};

export default EmploymentAdded;
