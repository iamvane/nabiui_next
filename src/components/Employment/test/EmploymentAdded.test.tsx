import * as React from 'react';
import EmploymentAdded from 'components/Employment/EmploymentAdded';
import { shallow } from 'enzyme';

describe('EmploymentAdded', () => {
  let wrapper: any;
  const deleteEmployment: () => void = jest.fn();
  const editEmployment: () => void = jest.fn();

  describe('When stillWorkHere is false', () => {
    beforeEach(() => {
      wrapper = shallow(
        <EmploymentAdded
          employer="Oracle"
          jobTitle="Software Engineer"
          jobLocation="Cambridge, MA, USA"
          fromMonth="sep"
          fromYear=""
          toMonth=""
          toYear="2018"
          stillWorkHere={false}
          deleteEmployment={deleteEmployment}
          editEmployment={editEmployment}
        />
      );
    });

    it('Matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('When stillWorkHere is true', () => {
    beforeEach(() => {
      wrapper = shallow(
        <EmploymentAdded
          employer="Oracle"
          jobTitle="Software Engineer"
          jobLocation="Cambridge, MA, USA"
          fromMonth="sep"
          fromYear="2017"
          toMonth="sep"
          toYear="2018"
          stillWorkHere={true}
          deleteEmployment={deleteEmployment}
          editEmployment={editEmployment}
        />
      );
    });

    it('Matches snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
