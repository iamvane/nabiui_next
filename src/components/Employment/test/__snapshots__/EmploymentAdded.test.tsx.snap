// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmploymentAdded When stillWorkHere is false Matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmploymentAdded
    deleteEmployment={[MockFunction]}
    editEmployment={[MockFunction]}
    employer="Oracle"
    fromMonth="sep"
    fromYear=""
    jobLocation="Cambridge, MA, USA"
    jobTitle="Software Engineer"
    stillWorkHere={false}
    toMonth=""
    toYear="2018"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div
          className="nabi-action-buttons"
        >
          <WithStyles(IconButton)
            aria-label="Delete"
            className="nabi-float-right nabi-margin-left-xsmall"
            color="primary"
            onClick={[Function]}
          >
            <WithStyles(Icon)>
              delete
            </WithStyles(Icon)>
          </WithStyles(IconButton)>
          <WithStyles(IconButton)
            aria-label="Edit"
            className="nabi-float-right"
            color="primary"
            onClick={[Function]}
          >
            <WithStyles(Icon)>
              edit
            </WithStyles(Icon)>
          </WithStyles(IconButton)>
        </div>,
        <WithStyles(Typography)
          className="nabi-margin-top-small nabi-text-uppercase"
          variant="body2"
        >
          <mark>
            Software Engineer
          </mark>
        </WithStyles(Typography)>,
        <WithStyles(Typography)>
          <mark>
            Oracle
          </mark>
        </WithStyles(Typography)>,
        <WithStyles(Typography)>
          <mark>
            Sep,  - , 2018
          </mark>
        </WithStyles(Typography)>,
        <WithStyles(Typography)>
          <mark>
            Cambridge, MA, USA
          </mark>
        </WithStyles(Typography)>,
      ],
      "className": "nabi-editable-item nabi-margin-top-small",
      "item": true,
      "md": undefined,
      "xs": 12,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <WithStyles(IconButton)
              aria-label="Delete"
              className="nabi-float-right nabi-margin-left-xsmall"
              color="primary"
              onClick={[Function]}
            >
              <WithStyles(Icon)>
                delete
              </WithStyles(Icon)>
            </WithStyles(IconButton)>,
            <WithStyles(IconButton)
              aria-label="Edit"
              className="nabi-float-right"
              color="primary"
              onClick={[Function]}
            >
              <WithStyles(Icon)>
                edit
              </WithStyles(Icon)>
            </WithStyles(IconButton)>,
          ],
          "className": "nabi-action-buttons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-label": "Delete",
              "children": <WithStyles(Icon)>
                delete
              </WithStyles(Icon)>,
              "className": "nabi-float-right nabi-margin-left-xsmall",
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "delete",
              },
              "ref": null,
              "rendered": "delete",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-label": "Edit",
              "children": <WithStyles(Icon)>
                edit
              </WithStyles(Icon)>,
              "className": "nabi-float-right",
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "edit",
              },
              "ref": null,
              "rendered": "edit",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <mark>
            Software Engineer
          </mark>,
          "className": "nabi-margin-top-small nabi-text-uppercase",
          "variant": "body2",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Software Engineer",
          },
          "ref": null,
          "rendered": "Software Engineer",
          "type": "mark",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <mark>
            Oracle
          </mark>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Oracle",
          },
          "ref": null,
          "rendered": "Oracle",
          "type": "mark",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <mark>
            Sep,  - , 2018
          </mark>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sep,  - , 2018",
          },
          "ref": null,
          "rendered": "Sep,  - , 2018",
          "type": "mark",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <mark>
            Cambridge, MA, USA
          </mark>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Cambridge, MA, USA",
          },
          "ref": null,
          "rendered": "Cambridge, MA, USA",
          "type": "mark",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="nabi-action-buttons"
          >
            <WithStyles(IconButton)
              aria-label="Delete"
              className="nabi-float-right nabi-margin-left-xsmall"
              color="primary"
              onClick={[Function]}
            >
              <WithStyles(Icon)>
                delete
              </WithStyles(Icon)>
            </WithStyles(IconButton)>
            <WithStyles(IconButton)
              aria-label="Edit"
              className="nabi-float-right"
              color="primary"
              onClick={[Function]}
            >
              <WithStyles(Icon)>
                edit
              </WithStyles(Icon)>
            </WithStyles(IconButton)>
          </div>,
          <WithStyles(Typography)
            className="nabi-margin-top-small nabi-text-uppercase"
            variant="body2"
          >
            <mark>
              Software Engineer
            </mark>
          </WithStyles(Typography)>,
          <WithStyles(Typography)>
            <mark>
              Oracle
            </mark>
          </WithStyles(Typography)>,
          <WithStyles(Typography)>
            <mark>
              Sep,  - , 2018
            </mark>
          </WithStyles(Typography)>,
          <WithStyles(Typography)>
            <mark>
              Cambridge, MA, USA
            </mark>
          </WithStyles(Typography)>,
        ],
        "className": "nabi-editable-item nabi-margin-top-small",
        "item": true,
        "md": undefined,
        "xs": 12,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <WithStyles(IconButton)
                aria-label="Delete"
                className="nabi-float-right nabi-margin-left-xsmall"
                color="primary"
                onClick={[Function]}
              >
                <WithStyles(Icon)>
                  delete
                </WithStyles(Icon)>
              </WithStyles(IconButton)>,
              <WithStyles(IconButton)
                aria-label="Edit"
                className="nabi-float-right"
                color="primary"
                onClick={[Function]}
              >
                <WithStyles(Icon)>
                  edit
                </WithStyles(Icon)>
              </WithStyles(IconButton)>,
            ],
            "className": "nabi-action-buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-label": "Delete",
                "children": <WithStyles(Icon)>
                  delete
                </WithStyles(Icon)>,
                "className": "nabi-float-right nabi-margin-left-xsmall",
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "delete",
                },
                "ref": null,
                "rendered": "delete",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-label": "Edit",
                "children": <WithStyles(Icon)>
                  edit
                </WithStyles(Icon)>,
                "className": "nabi-float-right",
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "edit",
                },
                "ref": null,
                "rendered": "edit",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <mark>
              Software Engineer
            </mark>,
            "className": "nabi-margin-top-small nabi-text-uppercase",
            "variant": "body2",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Software Engineer",
            },
            "ref": null,
            "rendered": "Software Engineer",
            "type": "mark",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <mark>
              Oracle
            </mark>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Oracle",
            },
            "ref": null,
            "rendered": "Oracle",
            "type": "mark",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <mark>
              Sep,  - , 2018
            </mark>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sep,  - , 2018",
            },
            "ref": null,
            "rendered": "Sep,  - , 2018",
            "type": "mark",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <mark>
              Cambridge, MA, USA
            </mark>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Cambridge, MA, USA",
            },
            "ref": null,
            "rendered": "Cambridge, MA, USA",
            "type": "mark",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`EmploymentAdded When stillWorkHere is true Matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <EmploymentAdded
    deleteEmployment={[MockFunction]}
    editEmployment={[MockFunction]}
    employer="Oracle"
    fromMonth="sep"
    fromYear="2017"
    jobLocation="Cambridge, MA, USA"
    jobTitle="Software Engineer"
    stillWorkHere={true}
    toMonth="sep"
    toYear="2018"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <div
          className="nabi-action-buttons"
        >
          <WithStyles(IconButton)
            aria-label="Delete"
            className="nabi-float-right nabi-margin-left-xsmall"
            color="primary"
            onClick={[Function]}
          >
            <WithStyles(Icon)>
              delete
            </WithStyles(Icon)>
          </WithStyles(IconButton)>
          <WithStyles(IconButton)
            aria-label="Edit"
            className="nabi-float-right"
            color="primary"
            onClick={[Function]}
          >
            <WithStyles(Icon)>
              edit
            </WithStyles(Icon)>
          </WithStyles(IconButton)>
        </div>,
        <WithStyles(Typography)
          className="nabi-margin-top-small nabi-text-uppercase"
          variant="body2"
        >
          <mark>
            Software Engineer
          </mark>
        </WithStyles(Typography)>,
        <WithStyles(Typography)>
          <mark>
            Oracle
          </mark>
        </WithStyles(Typography)>,
        <WithStyles(Typography)>
          <mark>
            Sep, 2017 - Present
          </mark>
        </WithStyles(Typography)>,
        <WithStyles(Typography)>
          <mark>
            Cambridge, MA, USA
          </mark>
        </WithStyles(Typography)>,
      ],
      "className": "nabi-editable-item nabi-margin-top-small",
      "item": true,
      "md": undefined,
      "xs": 12,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <WithStyles(IconButton)
              aria-label="Delete"
              className="nabi-float-right nabi-margin-left-xsmall"
              color="primary"
              onClick={[Function]}
            >
              <WithStyles(Icon)>
                delete
              </WithStyles(Icon)>
            </WithStyles(IconButton)>,
            <WithStyles(IconButton)
              aria-label="Edit"
              className="nabi-float-right"
              color="primary"
              onClick={[Function]}
            >
              <WithStyles(Icon)>
                edit
              </WithStyles(Icon)>
            </WithStyles(IconButton)>,
          ],
          "className": "nabi-action-buttons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-label": "Delete",
              "children": <WithStyles(Icon)>
                delete
              </WithStyles(Icon)>,
              "className": "nabi-float-right nabi-margin-left-xsmall",
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "delete",
              },
              "ref": null,
              "rendered": "delete",
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "aria-label": "Edit",
              "children": <WithStyles(Icon)>
                edit
              </WithStyles(Icon)>,
              "className": "nabi-float-right",
              "color": "primary",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": "edit",
              },
              "ref": null,
              "rendered": "edit",
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <mark>
            Software Engineer
          </mark>,
          "className": "nabi-margin-top-small nabi-text-uppercase",
          "variant": "body2",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Software Engineer",
          },
          "ref": null,
          "rendered": "Software Engineer",
          "type": "mark",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <mark>
            Oracle
          </mark>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Oracle",
          },
          "ref": null,
          "rendered": "Oracle",
          "type": "mark",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <mark>
            Sep, 2017 - Present
          </mark>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sep, 2017 - Present",
          },
          "ref": null,
          "rendered": "Sep, 2017 - Present",
          "type": "mark",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <mark>
            Cambridge, MA, USA
          </mark>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Cambridge, MA, USA",
          },
          "ref": null,
          "rendered": "Cambridge, MA, USA",
          "type": "mark",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <div
            className="nabi-action-buttons"
          >
            <WithStyles(IconButton)
              aria-label="Delete"
              className="nabi-float-right nabi-margin-left-xsmall"
              color="primary"
              onClick={[Function]}
            >
              <WithStyles(Icon)>
                delete
              </WithStyles(Icon)>
            </WithStyles(IconButton)>
            <WithStyles(IconButton)
              aria-label="Edit"
              className="nabi-float-right"
              color="primary"
              onClick={[Function]}
            >
              <WithStyles(Icon)>
                edit
              </WithStyles(Icon)>
            </WithStyles(IconButton)>
          </div>,
          <WithStyles(Typography)
            className="nabi-margin-top-small nabi-text-uppercase"
            variant="body2"
          >
            <mark>
              Software Engineer
            </mark>
          </WithStyles(Typography)>,
          <WithStyles(Typography)>
            <mark>
              Oracle
            </mark>
          </WithStyles(Typography)>,
          <WithStyles(Typography)>
            <mark>
              Sep, 2017 - Present
            </mark>
          </WithStyles(Typography)>,
          <WithStyles(Typography)>
            <mark>
              Cambridge, MA, USA
            </mark>
          </WithStyles(Typography)>,
        ],
        "className": "nabi-editable-item nabi-margin-top-small",
        "item": true,
        "md": undefined,
        "xs": 12,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <WithStyles(IconButton)
                aria-label="Delete"
                className="nabi-float-right nabi-margin-left-xsmall"
                color="primary"
                onClick={[Function]}
              >
                <WithStyles(Icon)>
                  delete
                </WithStyles(Icon)>
              </WithStyles(IconButton)>,
              <WithStyles(IconButton)
                aria-label="Edit"
                className="nabi-float-right"
                color="primary"
                onClick={[Function]}
              >
                <WithStyles(Icon)>
                  edit
                </WithStyles(Icon)>
              </WithStyles(IconButton)>,
            ],
            "className": "nabi-action-buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-label": "Delete",
                "children": <WithStyles(Icon)>
                  delete
                </WithStyles(Icon)>,
                "className": "nabi-float-right nabi-margin-left-xsmall",
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "delete",
                },
                "ref": null,
                "rendered": "delete",
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "aria-label": "Edit",
                "children": <WithStyles(Icon)>
                  edit
                </WithStyles(Icon)>,
                "className": "nabi-float-right",
                "color": "primary",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "children": "edit",
                },
                "ref": null,
                "rendered": "edit",
                "type": [Function],
              },
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <mark>
              Software Engineer
            </mark>,
            "className": "nabi-margin-top-small nabi-text-uppercase",
            "variant": "body2",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Software Engineer",
            },
            "ref": null,
            "rendered": "Software Engineer",
            "type": "mark",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <mark>
              Oracle
            </mark>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Oracle",
            },
            "ref": null,
            "rendered": "Oracle",
            "type": "mark",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <mark>
              Sep, 2017 - Present
            </mark>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sep, 2017 - Present",
            },
            "ref": null,
            "rendered": "Sep, 2017 - Present",
            "type": "mark",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <mark>
              Cambridge, MA, USA
            </mark>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Cambridge, MA, USA",
            },
            "ref": null,
            "rendered": "Cambridge, MA, USA",
            "type": "mark",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
